Documentation/Relationship/Relationship.md

```diff
@@ -0,0 +1,598 @@
+(in the example`writtenBy`) into the response or the query schema. Instead it is replaced by the defined relationship.    
+Other exporters should ideally rename such fields by adding a `Id` suffix (e.g. `writtenById`).
+
+**REST:**  
+Below is a example response generated from a REST exporter for such a `ArticleHandler`:
+
+```json
+{
+  "id": "3826",
+  "heading": "Retrieving Relationship Information",
+  "content": "...",
+  "_links": {
+    "author": "https://example.api/user/472",
+    "self": "https://example.api/article/3826"
+  }
+}
+```
+
+**GraphQL:**  
+A graphql query could look like the following:
+```graphql
+query {
+  article(id: "3826") {
+    heading
+    content
+    author {
+      id
+      name
+      lastname
+    }
+  }
+}
+```
+
+##### 3.3.2.2 Relationship Definition: `Inherits`
+
+A special case to the previous chapter is when you want to create a Relationship definition for the primary identifier
+of the data structure (e.g. the `.id` property of an `Identifiable`).  
+
+We have seen how this can work automatically in [3.3.1](#331-implicitly), when the return type of two `Handler` is the same.  
+But as explained, this isn't possible if the return types don't match up.
+
+Instead the user can explicitly define a `self` relationship. As we don't want the user to rely on any magic string constant
+(e.g. by defining `References<User>(at: \.id, as: "self")`) we introduce another relationship definition
+type `Inherits<User>(at: \.id)`. 
+Additionally a `References` definition MUST NOT have the reserved name `self`.  
+Similar to [3.3.1](#331-implicitly) such a `Inherits` definition will inherit all relationship information from the
+destination.
+
+The example below illustrates such a definition.
+
+```swift
+struct MeUser: Identifiable, WithRelationships {
+  var id: String
+  var loginToken: String
+
+  static var relationships: some RelationshipDefinition {
+    Inherits<User>(at: \.id)
+  }
+}
+// ...
+var content: some Component {
+  Group("user", $userId) {
+    UserHandler(userId: $userId)
+    // could contain other routes giving more information for the given user, e.g. ./posts/:postId
+  }
+  Group("me") {
+    MeUserHandler()
+  }
+}
+```
+
+**REST:**  
+A request to the `me` endpoint would then generate a response like the following:
+```json5
+{
+  "id": "532",
+  "loginToken": "rik0O1YK5wKjUY6CASjVRik0O1YK5wKjUY6CASjV",
+  "_links": {
+    "self": "https://example.api/user/532",
+    "post": "https://example.api/user/532/post/{postId}"
+  }
+}
+```
+
+**GraphQL:**  
+A graphql query for the given example might look like the following:
+
+[//]: # "// TODO graphql graphql must also be able to model the self link somehow. Can the properties of /user/:userId be inlined?"
+```graphql
+query {
+  me {
+    id
+    loginToken
+    post(postId: "123") {
+      title
+    }
+  }
+}
+```
+
+## 4. Appendix
+
+### 4.1. Example implementation of `var relationships`
```

Comment:
While this is an interesting detail and we should reuse the code I would suggest to put it in a swift file and merge it as part of the proposal or a separated PR so it is right in the project.

Reply:
Solved. See https://github.com/Apodini/Apodini/pull/41#issuecomment-748679123.